tutorials/cifar10-distributed.ipynb

```diff
@@ -0,0 +1,1319 @@
+        "\n",
+        "def run(backend=None, **spawn_kwargs):\n",
+        "    config[\"backend\"] = backend\n",
+        "    \n",
+        "    with idist.Parallel(backend=config[\"backend\"], **spawn_kwargs) as parallel:\n",
+        "        parallel.run(training, config)\n",
+        "\n",
+        "if __name__ == \"__main__\":\n",
+        "    fire.Fire({\"run\": run})"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_hMbi0-z-E-T"
+      },
+      "source": [
+        "The only change is how we run the script. We need to provide the IP address of the master node and its port along with the node rank. For example, for 2 nodes (`nnodes`) and 8 GPUs (`nproc_per_node`), we can:"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "rnOI--qIJ0ZN"
+      },
+      "source": [
+        "### Run with `torch.distributed.launch` (Recommended)\n",
+        "\n",
+        "On node 0 (master node):\n",
+        "\n",
+        "```\n",
+        "python -u -m torch.distributed.launch \\\n",
+        "    --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=0 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n",
+        "On node 1 (worker node):\n",
+        "\n",
+        "```\n",
+        "python -u -m torch.distributed.launch \\\n",
+        "    --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=1 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b1Z4JfvJJ6bt"
+      },
+      "source": [
+        "### Run with internal spawining\n",
+        "\n",
+        "On node 0:\n",
+        "```\n",
+        "python -u --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=0 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n",
+        "On node 1:\n",
+        "```\n",
+        "python -u --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=1 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KPUtXHI3KG9w"
+      },
+      "source": [
+        "### Run with horovodrun\n",
+        "\n",
+        "`np` below is calculated by `nnodes` x `nproc_per_node`.\n",
+        "\n",
+        "```\n",
+        "horovodrun -np 16 -H hostname1:8,hostname2:8 python -u main.py run --backend=\"horovod\"\n",
```

Comment:
```suggestion
        "horovodrun -np 4 -H hostname1:2,hostname2:2 python -u main.py run --backend=\"horovod\"\n",
```

Reply:
Actually, this is part of the example that I gave above - `The only change is how we run the script. We need to provide the IP address of the master node and its port along with the node rank. For example, for 2 nodes (nnodes) and 8 GPUs (nproc_per_node), we can:`
And therefore these values.

Reply:
I see. I just updated the values to something more realistic. Not everyone has 2 servers with 8 GPUs each

Reply:
Should I update the code for 2 nodes and 2 GPU's?

Reply:
Ok, 2 nodes with 2 GPUs each